### PR TITLE
Add an outline mode and use it both Site Editor and Template mode

### PR DESCRIPTION
**Pull request by youknowriad**
This PR moves the Adhoc outlines revealing implemented in edit-site into a dedicated mode of the block-editor package and uses that mode in both Site Editor and the template mode of the post editor.

### CODE REVIEW

**Comment by jasmussen** [comment]
There's a lot of CSS removed here, and when testing it I'm not seeing the exact same behavior of outlines in the site editor. The heuristic we decided to test for starters was:

- Hover = 1px blue
- Selected & focused = 1.5px blue
- Selected but not focused = 1px black

I'd like Jay who worked on this to test it a bit more indepthly in case I'm missing something, but it feels different.

This is purely an under the hood change, correct? Not surfaced in any prefs menu right?

**Comment by youknowriad** [comment]
Yes, under the hood and the removal of CSS is because for me it didn't feel like it deserved a dedicated style while we already had the style (the one we use for select mode)

**Comment by jameskoster** [comment]
I believe the per-state visual treatments Joen mentioned are an a11y requirement, so probably best to retain them. 

The only other thing I see missing is the `cursor: default` style on `.is-hovered`.

**Comment by jasmussen** [comment]
It's more a matter of us intentionally wanting to try out that new style as discussed in https://github.com/WordPress/gutenberg/pull/27271. While there's always room for improvement, it's barely been tested yet and I think it's way too early to decide it didn't work.

**Comment by youknowriad** [comment]
I can restore the styles but to be honest, I didn't even notice the difference myself so I really doubt we need any particular treatment here

**Comment by jasmussen** [comment]
I'm recording some demo work right now, and I've found a couple of things I want to tweak:

- Border radii aren't prefect
- The positioning is off at times
- #27296 is a bit annoying
- We might need to revisit the thicker select & focus style

But despite that, it feels like a really good balance that I think can be polished to work very well. I think we should keep it and refine it, and revisit in a few releases. Thank you Riad.

File: packages/block-editor/src/components/block-list/block.js [hunk 1]
```diff
@@ -122,6 +122,7 @@ function BlockListBlock( {
 				isDragging: isBlockBeingDragged( clientId ),
 				isHighlighted: isBlockHighlighted( clientId ),
 				isFocusMode: getSettings().focusMode,
+				isOutlineMode: getSettings().outlineMode,
```
Comment (ntsekouras) [inline packages/block-editor/src/components/block-list/block.js:125]: Nitpick - call `getSettings` once and extract props?

**Review by ntsekouras — APPROVED** [review]
Code LGTM. I don't see any visual change from master and works beautifully. Designers are better judges from me on the styling though...

Thanks Riad 👍

**Pull request merged**